lib/Onyx.js

```diff
@@ -378,13 +421,18 @@ function keysChanged(collectionKey, partialCollection) {
                 }
 
                 subscriber.withOnyxInstance.setState((prevState) => {
-                    const finalCollection = _.clone(prevState[subscriber.statePropertyName] || {});
+                    const prevCollection = prevState[subscriber.statePropertyName];
+                    const finalCollection = _.clone(prevCollection || {});
                     const dataKeys = _.keys(partialCollection);
                     for (let j = 0; j < dataKeys.length; j++) {
                         const dataKey = dataKeys[j];
                         finalCollection[dataKey] = cachedCollection[dataKey];
                     }
 
+                    if (deepEqual(prevCollection, finalCollection)) {
```

Comment:
Are there any performance implications on adding another deep equality check here?

Reply:
Good point and - there might be. Guides e.g. can have 10,000 reports that would exist in a "collection"

Reply:
removed it for now, lets discuss this in a separate issue / PR!